File: Patterns/9. Diamond Patterns/82. Hollow Full Diamond/hfd.py
n = int(input("Enter a Number : "))

for i in range(1,2*n):
    for j in range(1,2*n):
        if(i+j==n+1 or j-i==n-1 or i-j==n-1 or i+j==3*n-1):
            print("*",end=" ")
        else:
            print(end='  ')
    print()


'''
Output:

        *
      *   *
    *       *
  *           *
*               *
  *           *   
    *       *
      *   *       
        *          

Another approach using an extra vaiable row

  for i in range(1,2*n):
  
    row = min(i,2*n-i)

    for j in range(1,2*n):
      if(row+j==n+1 || j-row==n-1):
        print("*",end=" ")
      else:
        print(end='  ')
    print()

        
'''

File: Patterns/9. Diamond Patterns/82. Hollow Full Diamond/hfd.cpp
#include <iostream>
#include <algorithm>
using namespace std;
int main(){
    int n;
    cout<<"Enter a Number : ";
    cin>>n;

    for(int i=1; i<=2*n-1; i++){
        for(int j=1; j<=2*n-1; j++){
            if(i+j==n+1 || j-i==n-1 || i-j==n-1 || i+j==3*n-1){
                cout<<"* ";
            }else{
                cout<<"  ";
            }
        }
        cout<<'\n';
    }
}

/*
Output

        *         
      *   *       
    *       *     
  *           *   
*               * 
  *           *   
    *       *     
      *   *       
        * 

Another approach using an extra vaiable row

    for(int i=1; i<=2*n-1; i++){

        int row = min(i,2*n-i);

        for(int j=1; j<=2*n-1; j++){
            if(row+j==n+1 || j-row==n-1){
                cout<<"* ";
            }else{
                cout<<"  ";
            }
        }
        cout<<'\n';
    }

*/